<img src="images/unilogo.png" style="width: 200px;">

Paul Trap
===============================================
Authored by: [Nathan Belmore](https://plasma.physics.berkeley.edu/faculty-staff/nathan-belmore) 
In partnership with Tobias Tubandt

Published: 30.11.2018


# Abstract
***
In this experiment, we examined the mass of ion clouds confined in a paul trap. We determine the mass of the ion species through 3 methods. First, we look at time of flight duing extraction. Second, we excite the ion clouds using an additional drive frequency. When the approaching the resonant frequency of a given ion species the ions will be driven out of confinment. Lastly, we sweep the amplitude of the confinement field. Since the stability of confinement is also dependent on mass, over the sweep, lighter particles will be driven out of confinement first.  

# Physical Principles
***
To contain a cloud of ions we'll need a force that creates something akin to a 3D harmonic potential, i.e. $F = -k\vec{r}$ where $k$ is a positive constant. Since the electric field is proportional to the force we can expect the potential to take the form

$$\Phi \propto \alpha x^2 + \beta y^2 + \gamma z^2$$

Enforcing the condition $\Delta \Phi = 0$ we get that $\alpha + \beta + \gamma = 0$, which has multipule solutions. However, we can pick the canonical choice of $\alpha = \beta$, and $\gamma = -2 \alpha$. This results in:

$$\Phi(x,y,z) = \frac{\Phi_0}{\gamma} (x^2 + y^2 - 2 z^2)$$

Where $\frac{\Phi_0}{\gamma}$ acts as our proportionality constant. This results in a hpyerbolic potential structure. We can apply this to the traditioal paul trap design and shift rewrite the prefactor in terms of the geometery of the trap. 

| <img src="images/trap.png" style="width: 400px;">|
|----------------------------------------------------------------------------|
| Fig. 1 Generic 3D paul trap design. [1] |

We can define $d_0 = \frac{r_0^2}{2} + z_0$ and $\gamma = 2 d_0$. In an ideal paul trap we can decompose the applied field into $U_0$ and $V_0 \cdot cos(\Omega t)$, a constant and time dependent field. Inserting this expression for $\Phi_0$ we get the following: 

$$\Phi(x,y,z) = \frac{U_0 + V_0 \cdot \cos(\Omega t)}{2 d_0} (x^2 + y^2 - 2 z^2)$$

Solving for the cylindrical EOM:

$$\ddot{r} + \frac{e}{m 2d_0^2}(U - V cos(\omega t))r = 0$$

$$\ddot{z} + \frac{e}{m 2d_0^2}(U - V cos(\omega t))z = 0$$

Using the following substitutions $a_z = -2 a_r = -\frac{4eU}{m d_0^2 \Omega^2}$, $q_z = -2 q_r = -\frac{4eV}{m d_0^2 \Omega^2}$, and $\tau = \frac{1}{2} \Omega t$ give us:

$$0 = $$

$$\frac{d^2 r}{d \tau^2} + (a_r - 2 q_r cos(2 \tau))r = 0$$

$$\frac{d^2 z}{d \tau^2} + (a_z - 2 q_z cos(2 \tau))z = 0$$

The solution of which has stable and unstable regions. This is a named problem know as Mathieu’s equation. The stable solutions form a manifold in $a$ $q$ space, which represents the solutions that produce stable confinment. 

| <img src="images/stability.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 2 Mathieu Stability Diagram for Nitrogen |


## Execution and Setup
***
### Setup
We used a 3D hyperbolical Paul trap, the geometery of which can be found in Fig. 3. In the experiement we use 2 end caps as electrodes and a ring electrode to induce the wanted potential. We apply a trapping RF field using a function generator that is amplifed. There is a second function generator for applying an additional field on top of the trapping field. 

In fig. 3 is the schematic of a Paul trap. The space in which the ions can move, can be described as a cylinder with hyperbolic borders. We use a vacuum chamber and an electrode gun to ionize the atoms and molecules in the chamber. To get an actual measurement the ions are lead through a channeltron to get a signal. Through the different fly times of the ions different masses can be measured.

| <img src="images/diagram.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 3 Diagram of the Paul trap |

# Analysis
***



In [1]:
# creating the correct enviroment for analysis
import os
import numpy as np
import pandas as pd
import scipy.constants as c
from scipy.optimize import curve_fit
from bqplot import pyplot as plt
from bqplot import *
from bqplot.interacts import BrushIntervalSelector
from ipywidgets import ToggleButtons, VBox, HTML

## Importing the data

Some of the data sets contain multipule itterations, which are summed over to improve the statistics. 

In [2]:
# path to the data set
path = 'data/'

# setup to scan just the first level of folders in the data set
# first step is to produce a list of the folders that contain each data set
folders = next( os.walk(path) )[1]

# sorting them to control the order
folders.sort() 

# some data structure for genetrating a list of the files. 
files = [[],[],[],[],[],[],[],[]]
list = []


# a loop to generate a list of files for each folder
for i in range(0, len(folders)):
    list.append(os.listdir(path + folders[i]))
    print(path + folders[i])
    for file in list[i]:
        if file.endswith(".dat"):
            files[i].append(os.path.join(path + folders[i], file))
    files[i].sort()

    
# importing the data in a pandas dataframe

# data set 5    
df_list = [pd.read_table(file, usecols=[1]) for file in files[3]]
data5 = pd.concat(df_list, axis=1, ignore_index=True)

# data set 9
df_list = [pd.read_table(file, usecols=[1]) for file in files[7]]
data9 = pd.concat(df_list, axis=1, ignore_index=True)
# number of data points
n = 89 
# summing multiple iterations
data9 = pd.concat([data9[data9.columns[0+i:-1:n]].sum(axis=1) for i in range(n+1)], axis=1)
    
# data set 12    
df_list = [pd.read_table(file, usecols=[1]) for file in files[2]]
data12 = pd.concat(df_list, axis=1, ignore_index=True)
# number of data points
n = 499 
# summing multiple iterations
data12 = pd.concat([data12[data12.columns[0+i:-1:n]].sum(axis=1) for i in range(n+1)], axis=1)



# Code for importing other data sets, but we don't use them so it makes no sense to import them. 

# df_list = [pd.read_table(file, usecols=[1]) for file in files[0]]
# data10 = pd.concat(df_list, axis=1, ignore_index=True)

# df_list = [pd.read_table(file, usecols=[1]) for file in files[1]]
# data11 = pd.concat(df_list, axis=1, ignore_index=True)

# df_list = [pd.read_table(file, usecols=[1]) for file in files[4]]
# data6 = pd.concat(df_list, axis=1, ignore_index=True)

# df_list = [pd.read_table(file, usecols=[1]) for file in files[5]]
# data7 = pd.concat(df_list, axis=1, ignore_index=True)

# df_list = [pd.read_table(file, usecols=[1]) for file in files[6]]
# data8 = pd.concat(df_list, axis=1, ignore_index=True)

data/10
data/11
data/12
data/5
data/6
data/7
data/8
data/9


### Dataset 5

For this dataset, we applied a driving field of $430 khz$ with an amplitude of $14.6V$. We scanned over an externally applied frequency which was varied from $10khz - 500khz$ in $500hz$ steps. 

We applied the external field to drive oscillations in the trapped particles. If the particles are resonant with the drive frequency, we might expect to see the particles driven out of confinement. Since we expect this trap configuration to have several species of particles, there will likely be several peaks with distinct resonant frequencies. 

We'll start with a simple approach. Sum all the data points in each data set and plot it as a function of the data set. Since the data sets are divided by applied frequency, we get a plot of Applied Frequency vs. Integrated Counts. 

In [32]:
row, col = data5.shape

# sum each data set
y = [data5[i].sum() for i in range(0,col)]
# generating the x-axis, 10khz - 500khz with the number of steps = col
x = np.linspace(10,500,col)

# splot the results
fig=plt.figure()
line = plt.plot(x, y, marker='circle', stroke_width=.5, opacities = [.5])
fig.layout.height = '500px'
fig.layout.width = '800px'
plt.xlabel('Applied frequency (khz)')
plt.ylabel('Total count at fixed applied frequency')
plt.title('Applied frequency sweep')
plt.show()

As can be seen above, there is a lot of noise in the signal. We are also summing over all the species, so there exists multiple resonances overlapping in the data set. Instead of integrating over the whole dataset we can focus our integration region over a subset of the data that corresponds to the peaks, where we expect a significant change to do the applied frequency. Below we'll create a function for selecting subsets of the data.

In [4]:
# To speed up the analysis we'll create a fuction for selecting subsets of the data.

def findminmax(values):
    """Returns an array of [min, max] of the values array"""
    if values == []:
        return [0, 0]
    else:
        return [np.min(values),np.max(values)]

def selectPlot(x, y, color, xlabel, ylabel, title):
    # The Mark we want to select subsamples of
    scales = {'x': LinearScale(), 'y': LogScale()}
    global scatter
    scatter = Scatter(x=x, y=y, scales=scales, colors=[color],
                      selected_style={'opacity': '1'}, unselected_style={'opacity': '0.2'})
    # Create the brush selector, passing it its corresponding scale.
    # Notice that we do not pass it any marks for now
    brushintsel = BrushIntervalSelector(scale=scales['x'])

    x_ax = Axis(label=xlabel, scale=scales['x'])
    x_ay = Axis(label=ylabel, scale=scales['y'], orientation='vertical')
    # Pass the Selector instance to the Figure
    fig = Figure(marks=[scatter], axes=[x_ax, x_ay],
                 title=title,
                 interaction=brushintsel)

    # The following text widgets are used to display the `selected` attributes
    text_brush = HTML()
    text_scatter = HTML()

    # This function updates the text, triggered by a change in the selector
    def update_scatter_text(*args):
        """Fuction used to update the text indicating the selected range"""
        text_scatter.value = "The scatter's selected indices are {} through {}".format(findminmax(scatter.selected)[0], findminmax(scatter.selected)[1])
    scatter.observe(update_scatter_text, 'selected')

    update_scatter_text()

    # Display
    brushintsel.marks = [scatter]
    return VBox([fig, text_brush, text_scatter])

### Example of the peak analysis process

The function integrates the data set as a function of the arrival times of each set. Looking at the results can give us a sense of where the particles peaks are located in the data sets and allow us to narrow the integration range. By selecting a range of values on the graph, we can pass the selected range to the next cell for analysis.

In [25]:
y1 = data5[data5.columns[0:-1]][:300].sum(axis=1).values
x1 = np.multiply(range(0,len(y1)),0.1)

selectPlot(x1, y1, 'orange', 'Arrival time (microseconds)', 'Intensity (int. counts)', '''Integration Interval Selector. Click and drag on the Figure to select window.''')

Now we can use that range to define the integration region and re-plot our first plot over the narrow region. Since I want to select a region to integrate over, I'll call the selectPlot function again on the new range. However, we are now integrating each data set over the interval of arrival times defined by the window above so that the output will be a function of applied drive frequency (like the first plot). 

In [26]:
row, col = data5.shape
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
y2 = [data5[i][indexrange[0]:indexrange[1]].sum() for i in range(0,col)]
x2 = np.linspace(10,500,col)

selectPlot(x2, y2, 'blue', 'Applied frequency (khz)', 'Total count at fixed applied frequency', '''Applied frequency sweep''')

The range selected from above at the time of running the cell is : 88 to 96


Now we see a clear resonance peak around 100hz (compared to the first plot). We can fit the peak to get a sense of the peak value. At the same time we fit the arrival time peak to get the corresponding arrival time.  

In [27]:
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))

yfit = y2[np.min(scatter.selected):np.max(scatter.selected)]
xfit = x2[np.min(scatter.selected):np.max(scatter.selected)]

# Because of the noise on the data it'll be easier to fit a simple fuction.
# This is fine since we are just looking for a center point of the peak.
def function(x, x0, a, b):
    return a*(x-x0)**2+b

# Fitting and plotting the results
popt1, pcov = curve_fit(function, xfit, yfit)

fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(x2, y2, marker='circle', stroke_width=.5, opacities = [.5])
fit = plt.plot(xfit, function(xfit, *popt1), colors=['green'])
plt.xlabel('Applied frequency (khz)')
plt.ylabel('Total count at fixed applied frequency')
plt.title('Applied frequency sweep')
plt.show()

print("The fitted frequency peak is {:5.5}khz".format(popt1[0]))


# I also want to get the corosponding peak value of the applied frequency
yfit2 = y1[indexrange[0]:indexrange[1]]
xfit2 = x1[indexrange[0]:indexrange[1]]

# Fitting and plotting the results
p0 = [np.mean(xfit2), -1, np.max(xfit2)]
popt1, pcov = curve_fit(function, xfit2, yfit2, p0 = p0)
fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
# fit = plt.plot(xfit2, yfit2, marker='circle', colors=['orange'], stroke_width=.5, opacities = [.5])
fit = plt.plot(x1[20:250], y1[20:250], marker='circle', colors=['orange'], stroke_width=.5, opacities = [.5])
fit = plt.plot(xfit2, function(xfit2, *popt1), colors=['green'])
plt.xlabel('Time in microseconds')
plt.ylabel('Total count at fixed applied frequency')
plt.title('Applied frequency sweep')
plt.show()

print("The corresponding mass peak is at {:5.5} microseconds".format(popt1[0]))

The range selected from above at the time of running the cell is : 88 to 96


The fitted frequency peak is 100.6khz


The corresponding mass peak is at 9.1992 microseconds


Repeating the process with the other peaks in the data set 5 provides the results in:

## Other peaks

We repeated the above process to produce the data for the 2 addtional peaks.

| Peaks  | $\omega$ (khz) | $\mu$ s |
|--------|----------|---------|
| Peak 1 | 100.6    | 9.21   |
| Peak 2 | 72.6     | 10.45  |
| Peak 3 | 47.5     | 12.41  |


| <img src="images/10mus.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 4  Second peak fit (10.45 $\mu$ s) |

| <img src="images/71khz.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 5  Second peak fit (72.6 khz) |


| <img src="images/12mus.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 6  Third peak fit (12.41 $\mu$ s) |

| <img src="images/45khz.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 7  Third peak fit (47.5 khz) |

As expected, the particles with higher arrival times have lower resonance frequencies (larger mass). 

### Dataset 12

Dataset 12 is similar to dataset 5 but the trapping was optimized for a single species. We expect one strong signal compared to the many weak signals in dataset 5. 

In [28]:
y1 = data12[data12.columns[0:-1]][:300].sum(axis=1).values
x1 = np.multiply(range(0,len(y1)),0.1)

selectPlot(x1, y1, 'orange', 'Arrival time (nanooseconds)', 'Intensity (int. counts)', '''Integration Interval Selector. Click and drag on the Figure to select window.''')

In [29]:
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]

print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
row, col = data12.shape
y2 = [data12[i][indexrange[0]:indexrange[1]].sum() for i in range(0,col)]
x2 = range(0,499)

selectPlot(x2, y2, 'blue', 'Applied frequency (khz)', 'Total count at fixed applied frequency', '''Applied frequency sweep''')

The range selected from above at the time of running the cell is : 66 to 89


In [30]:
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))

yfit = y2[np.min(scatter.selected):np.max(scatter.selected)]
xfit = x2[np.min(scatter.selected):np.max(scatter.selected)]

# Because of the noise on the data it'll be easier to fit a simple fuction.
# This is fine since we are just looking for a center point of the peak.
def function(x, x0, a, b):
    return a*(x-x0)**2+b

# Fitting and plotting the results
p0 = [np.mean(xfit), 1, np.max(xfit)]
popt1, pcov = curve_fit(function, xfit, yfit, p0 = p0)

fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(x2, y2, marker='circle', stroke_width=.5, opacities = [.5])
fit = plt.plot(xfit, function(xfit, *popt1), colors=['green'])
plt.xlabel('Applied frequency (khz)')
plt.ylabel('Total count at fixed applied frequency')
plt.title('Applied frequency sweep')
plt.show()

print("The fitted frequency peak is {:5.5}khz".format(popt1[0]))


# I also want to get the corosponding peak value of the applied frequency
yfit2 = y1[indexrange[0]:indexrange[1]]
xfit2 = x1[indexrange[0]:indexrange[1]]

# Fitting and plotting the results
p0 = [np.mean(xfit2), -1, np.max(xfit2)]
popt1, pcov = curve_fit(function, xfit2, yfit2, p0 = p0)
fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(x1[0:250], y1[0:250], marker='circle', colors=['orange'], stroke_width=.5, opacities = [.5])
fit = plt.plot(xfit2, function(xfit2, *popt1), colors=['green'])
plt.xlabel('Arrival time (microseconds)')
plt.ylabel('Total count from data set')
plt.title('Total count vs. Arrival time')
plt.show()

print("The corresponding mass peak is at {:5.5} microseconds".format(popt1[0]))

The range selected from above at the time of running the cell is : 66 to 89


The fitted frequency peak is 119.31khz


The corresponding mass peak is at 7.5885 microseconds


## Other peaks

We repeated the above process to produce the data for both peaks.

| Peaks  | $\omega$ (khz) | $\mu$ s |
|--------|----------|---------|
| Peak 1 | 119.50   | 7.57 |
| Peak 2 | 59.08    | 7.57 |



| <img src="images/58khz.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 8  First peak fit (59.08 khz) |

| <img src="images/119khz.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 9  Second peak fit (119.50 khz) |

### Data Set 9 - Amplitude Sweep 



In [31]:
y1 = data9[data9.columns[0:-1]][:300].sum(axis=1).values
x1 = np.multiply(range(0,len(y1)),0.1)

selectPlot(x1[0:300], y1[0:300], 'orange', 'Arrival time (microseconds)', 'Intensity (int. counts)', '''Integration Interval Selector. Click and drag on the Figure to select window.''')

In [8]:
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
row, col = data9.shape
y2 = [data9[i][indexrange[0]:indexrange[1]].sum() for i in range(0,89)]

The range selected from above at the time of running the cell is : 68 to 85


In [10]:
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
y3 = [data9[i][indexrange[0]:indexrange[1]].sum() for i in range(0,89)]

The range selected from above at the time of running the cell is : 87 to 101


In [11]:
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
y4 = [data9[i][indexrange[0]:indexrange[1]].sum() for i in range(0,89)]

The range selected from above at the time of running the cell is : 107 to 117


In [12]:
indexrange = [np.min(scatter.selected), np.max(scatter.selected)]
print('The range selected from above at the time of running the cell is : ' + str(indexrange[0]) + ' to ' + str(indexrange[1]))
y5 = [data9[i][indexrange[0]:indexrange[1]].sum() for i in range(0,89)]

The range selected from above at the time of running the cell is : 110 to 120


In [13]:
x2 = np.linspace(0.1,8,89)

fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(x2, y2, marker='circle', colors=['red'], stroke_width=.5, opacities = [.5])
fit = plt.plot(x2, y3, marker='circle', colors=['blue'], stroke_width=.5, opacities = [.5])
fit = plt.plot(x2, y4, marker='circle', colors=['green'], stroke_width=.5, opacities = [.5])
fit = plt.plot(x2, y5, marker='circle', colors=['orange'], stroke_width=.5, opacities = [.5])
plt.xlabel('Confinement Amplitude (V)')
plt.ylabel('Total count at fixed drive amplitude')
plt.title('Confinement voltage sweep')
plt.show()

The four colored plots above corospond to the 4 intervals below:

| <img src="images/peaks.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 10 Arrival time plot for 4 intervals |

### Voltage Calibration 

While we recorded the applied confinement amplitude, the signal goes through an amplifier. We need to convert the applied voltage. 

In [14]:
vin = [0.1, 1, 2, 3, 4, 5, 6, 7, 8, 9]
vout = [0.5, 6.5, 11, 14.9, 20, 25, 28.5, 30, 31, 32.5]

fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(vin, vout, marker='triangle-up', colors=['red'], stroke_width=.5, opacities = [.5])
plt.xlabel('Confinement Amplitude (V)')
plt.ylabel('Total count at fixed drive amplitude')
plt.title('Confinement voltage sweep')
plt.show()

def convertV(v):
    """Converts the applied voltage to the voltage seen after the amplifier."""
    return np.interp(v, vin, vout)

Now we can plot the converted amplitude.

In [15]:
# converting the x-axis
xnew = convertV(x2)

# re-plotting the graph
fig = plt.figure()
fig.layout.height = '500px'
fig.layout.width = '800px'
fit = plt.plot(xnew, y2, marker='circle', colors=['red'], stroke_width=.5, opacities = [.5])
fit = plt.plot(xnew, y3, marker='circle', colors=['blue'], stroke_width=.5, opacities = [.5])
fit = plt.plot(xnew, y4, marker='circle', colors=['green'], stroke_width=.5, opacities = [.5])
fit = plt.plot(xnew, y5, marker='circle', colors=['orange'], stroke_width=.5, opacities = [.5])
plt.xlabel('Confinement Amplitude (V)')
plt.ylabel('Total count at fixed drive amplitude')
plt.title('Corrected Confinement voltage sweep')
plt.show()

### Mass Conversion (DS5)

Since this data contains multipule species we can use a simple method to check the possible masses. If we expect the largest inital signal to be from nitrogen (first peak), we can check our expectations for the later peaks.

| Peaks  | $\mu$ s | closest mass | Expected Value  | $\delta$  |
|--------|----------|---------|---------|---------|
| Peak 1 | 9.205 | nitrogen | 9.205 | 0 |
| Peak 2 | 10.45 | oxygen | 10.55 | 0.1 |
| Peak 3 | 12.41 | 18-19 AMU | ??? | ??? |

The signal from the third peak is likely a mesh of signals from several species. But the ratios for nitrogen and oxygen seem good. We can further compare the data peaks using the Mathieu stability diagrams.

| <img src="images/stability.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 11 Mathieu Stability Diagram for Nitrogen |

| Peaks  | $\omega$ (khz) | Expected Value  | $\delta$  |
|--------|----------|---------|---------|
| Peak 1 | 100.6 | 89.90 | 10.7 |
| Peak 2 | 72.6 | 77.2 | 5.4 |

The results of this dataset aren't clear, but it was a small sample and there were many species. Let's examine dataset 12, which should better adress this issue. 


### Mass Conversion (DS12)

| <img src="images/stabilitydata12.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 12 Mathieu Stability Diagram for Nitrogen |


| Peaks  | $\omega$ (khz) | $\mu$ s | Expected Value  | $\delta$  |
|--------|----------|---------|---------|---------|
| Peak 1 | 119.50 | 7.57 | 119.00 | 0.5 |
| Peak 2 | 59.08 | 7.57 | 55.38 | 3.7 |

As you can see, not only did we find the $z_{sec}$ resonance for nitrogen, we see the $r_{sec}$ one and higher order modes in the data set. Tuning the cavity and larger statistics helped a great deal. We can also look at the final method to assess mass.


### Mass Conversion (DS9)

| <img src="images/peaks.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 13 Arrival time plot for 4 intervals |

| <img src="images/Vsweep.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 14 Voltage vs. count for the 4 intervals |

From fig 13 and fig 14, we are farily confident the first peak is nitrogen in the data set. Tracking the red line in the corrected confinement voltage sweep suggests that around 22V amplitude we see a dropoff in nitrogen gas. Unfortunately, this isn't reflected by the Mathieu stability chart. 


| <img src="images/stabilitydata9.png" style="width: 600px;">|
|----------------------------------------------------------------------------|
| Fig. 15 Mathieu Stability Diagram for Nitrogen (@22V V_rf) |


It seems like the signal of DS9 was reasonably well defined, but we weren't able to related it to the diagrams in the end. 

## Discussion and Conclusion
***
As expected, the nitrogen signal was very strong and with more statistics it might be possible to clarify the issues with dataset5. We expect that much of the problem was overlapping peaks which could not be discriminated using the resolution of the detector. We very accuratly measured the resonant frequencies of nitrogen of $119.50 khz$ and $59.08 khz$ and managed to identify both nitrogen and oxygen in the dataset. 

## References

[1] Werth, G., Gheorghe, V. N. and Major, F. G., ‘Charged Particle Traps II: Applications ‘, Springer, Berlin, (2009), p.4.